transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -279,34 +295,44 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
           case Some(schema) => {
             // if the schema is not able to load, throw an exception, otherwise try to use it to validate
             Schema.loadFromString(schema) match {
-              case Failure(_) =>
-                throw new Exception(
-                  s"Schema not loaded properly for schema at $url, file $filename"
-                )
+              case Failure(_) => {
+                val errorMessage = s"Schema not loaded properly for schema at $url, file $filename"
+                logSchemaValidationError(filename, errorMessage)
```

Comment:
I don't know much if anything about this code at this point, but my usual pattern for code that throws and logs is to log at the catch point, not the throw point. That way you don't have to remember to add a log to every throw. Plus, making sure you log at catch points means that exceptions are always logged.

It looks like in this case you want something special to be logged for this specific exception, in that case throwing a custom object that either itself knows how to log, or can be used to determine a special logging path at the catch point, would let you log this exception differently than others.

Reply:
This is a very good point; this is my bad for designing it the way I did without an actual catching piece, and this reference material might help with fixing it up https://docs.scala-lang.org/overviews/scala-book/functional-error-handling.html

Reply:
Thank you! I agree, that catching it at a single point sounds like the better way to go. I'll go ahead and update it.

Reply:
I updated it to use a try catch but haven't updated the corresponding test yet (it's still expecting an error to be thrown). Maybe we can talk a bit more about what we want the behavior to look like here when the three of us meet this afternoon?